pkg/rancher-components/src/components/Form/ArrayList/ArrayList.vue

```diff
@@ -114,115 +118,117 @@ export default {
   },
 
   computed: {
-    isView() {
+    isView(): boolean {
       return this.mode === _VIEW;
     },
 
-    showAdd() {
+    showAdd(): boolean {
       return this.addAllowed;
     },
 
-    showRemove() {
+    showRemove(): boolean {
       return this.removeAllowed;
     },
 
-    isDefaultProtip() {
+    isDefaultProtip(): boolean {
       return this.protip === DEFAULT_PROTIP;
     },
 
-    showProtip() {
+    showProtip(): boolean {
       if (this.protip && !this.isDefaultProtip) {
         return true;
       }
 
-      return !this.valueMultiline && this.protip;
-    }
+      return !this.valueMultiline && !!this.protip;
+    },
   },
 
   watch: {
     value() {
       this.lastUpdateWasFromValue = true;
-      this.rows = (this.value || []).map(v => ({ value: v }));
+      this.rows = (this.value || []).map((v) => ({ value: v }));
     },
     rows: {
       deep: true,
-      handler(newValue, oldValue) {
+      handler() {
         // lastUpdateWasFromValue is used to break a cycle where when rows are updated
         // this was called which then forced rows to updated again
         if (!this.lastUpdateWasFromValue) {
           this.queueUpdate();
         }
         this.lastUpdateWasFromValue = false;
-      }
-    }
+      },
+    },
   },
 
   created() {
-    this.queueUpdate = debounce(this.update, 50);
+    debounce(this.update, 50);
   },
 
   methods: {
+    queueUpdate(){
```

Comment:
It has been pointed out that this would create issue in case of multiple components it would cause issues.

Reply:
Refactored to remove queueUpdate